### PR TITLE
Refactor the progress tracker to be non-asset-specific

### PR DESCRIPTION
**Pull request by volkangurel**
Refactor the progress tracker to be non-asset-specific and allow further refactoring of executable entrypoints

### CODE REVIEW

File: layer/main/asset.py [hunk 1]
```diff
@@ -249,34 +244,14 @@ def _ui_progress_with_tracker(
...
-            tracker.mark_model_training(asset_name)
-        elif asset_type == AssetType.DATASET:
-            tracker.mark_dataset_building(asset_name)
-    elif asset_type == AssetType.MODEL:
-        tracker.mark_model_loaded(asset_name)
-    elif asset_type == AssetType.DATASET:
-        tracker.mark_dataset_loaded(asset_name)
+        tracker.mark_running(asset_type, asset_name)
+    else:
+        tracker.mark_asset_downloaded(asset_type, asset_name)
```
Comment (adrien-layer) [inline layer/main/asset.py:254]: other one is `mark_uploading` so maybe either this one is `mark_downloaded` or other one is `mark_asset_uploading`?

**Review by adrien-layer — APPROVED** [review]
minor comment

**Review by cihati — COMMENTED** [review]
I think @aleksmitov should review this before we merge

File: layer/main/asset.py (hunk 1 above)
Comment (volkangurel) [reply layer/main/asset.py:254]: good point, done!

**Review by aleksmitov — APPROVED** [review]
Looks good to me, just please make exhaustive enough manual tests to ensure there are no UI regressions.

**Comment by volkangurel** [comment]
Thanks @aleksmitov, already done!

**Pull request merged**